lots.py

```diff
@@ -1,18 +1,71 @@
 import random
+import numpy as np
 
 
-possible_castings = ['|', '/', '\\', '_', '-']
+# Keys store casting possibilities
+# First list entry is the cost in lots (how many lots to construct the cast)
+# Second list entry is the relative probability
+# Purposely omitted: V, ^, N, W, T, L, H, A, [, ]
+castings = {
+    '|': [1, 1],
+    '/': [1, 1],
+    '\\': [1, 1],
+    '_': [1, 1],
+    '-': [1, 1],
+    '<': [2, 0.15],
+    '>': [2, 0.15],
+    '+': [2, 0.1],
+    '=': [2, 0.1],
+    'X': [2, 0.1],
+    'Z': [3, 0.02],
+    'Y': [3, 0.02],
+    'E': [4, 0.02],
+    '#': [4, 0.001],
+}
+
+# Normalize probabilities
+probabilities = [castings[cast][1] for cast in castings]
+probabilities = [i / sum(probabilities) for i in probabilities]
+
 
 class Lots:
-    def __init__(self, num_lots=5):
+    def __init__(self, num_lots=10):
+        if num_lots < 1 or not isinstance(num_lots, int):
+            raise ValueError()
```

Comment:
be more descriptive with this error:

`raise ValueError('num_lots must be an integer greater than 1')`